homeassistant/components/garmin_connect/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Garmin Connect integration."""
+import logging
+
+from garminconnect import (
+    Garmin,
+    GarminConnectAuthenticationError,
+    GarminConnectConnectionError,
+    GarminConnectTooManyRequestsError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class GarminConnectConfigFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Garmin Connect."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def _show_setup_form(self, errors=None):
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return await self._show_setup_form(user_input)
+
+        garmin_client = Garmin(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+
+        errors = {}
+        try:
+            garmin_client.login()
+        except GarminConnectConnectionError:
+            errors["base"] = "cannot_connect"
+            return await self._show_setup_form(errors)
+        except GarminConnectAuthenticationError:
+            errors["base"] = "invalid_auth"
+            return await self._show_setup_form(errors)
+        except GarminConnectTooManyRequestsError:
+            errors["base"] = "too_many_requests"
+            return await self._show_setup_form(errors)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+            return await self._show_setup_form(errors)
+
+        unique_id = garmin_client.get_full_name()
+
+        entries = self._async_current_entries()
+        for entry in entries:
+            if entry.data[CONF_ID] == unique_id:
```

Comment:
Please set a unique id instead. Use `self.async_set_unique_id` and then `self._abort_if_unique_id_configured`.

https://developers.home-assistant.io/docs/en/config_entries_config_flow_handler.html#unique-ids

https://github.com/home-assistant/home-assistant/blob/aa390efd694dea5ea62319621959120218b3ee06/homeassistant/config_entries.py#L819-L853

Reply:
You mean replace it by those two calls?
This will result in a 'Aborted' message only, without any explanation why.
Not really an user experience improvement.
How to get this functionality back? (reason="already_setup") 

```

        await self.async_set_unique_id(unique_id)
        self._abort_if_unique_id_configured()

        # entries = self._async_current_entries()
        # for entry in entries:
        #     if entry.data[CONF_ID] == unique_id:
        #         return self.async_abort(reason="already_setup")

```

Reply:
Fill in the corresponding string with the message.

Reply:
It's the `"already_configured"` string that we need to set in strings.json.